### PR TITLE
Layout fixes for tab bar and tab stack panel

### PR DESCRIPTION
**Pull request by ellisonbg**
This brings up the core layout of the dock area to where it was before the mono-repo. I am also using less to better parametrize the border/width/height of the tabs.

### CODE REVIEW

**Comment by ellisonbg** [comment]
I should mention that I have a lot more work like this on the way...

Here is a screenshot of what this looks like:

screen shot 2016-08-25 at 6 08 13 pm

File: src/basestyle/phosphor.less [hunk 1]
```diff
@@ -168,15 +172,16 @@
 /*-----------------------------------------------------------------------------
 | TabBar
 |----------------------------------------------------------------------------*/
+
 .p-TabBar {
   overflow: visible;
   color: #777777;
-  font-size: 13px;
+  font-size: @jp-ui-font-size;
```
Comment (blink1073) [inline src/basestyle/phosphor.less]: Should we follow our CSS naming convention and use `jp-ui-fontSize` here?

File: src/basestyle/phosphor.less [hunk 2]
```diff
@@ -386,15 +393,17 @@
 /*-----------------------------------------------------------------------------
 | DockPanel
 |----------------------------------------------------------------------------*/
+
 .p-DockPanel-tabPanel {
+  // This padding is needed to shwo the shadow on .p-TabPanel-stackedPanel
```
Comment (blink1073) [inline src/basestyle/phosphor.less]: show (sp.)

**Comment by blink1073** [comment]
Should this explicitly close #680?

File: src/basestyle/phosphor.less (hunk 1 above)
Comment (ellisonbg) [inline src/basestyle/phosphor.less]: Thought about that - the main difference is that things like `font-size` are actual CSS properties. Many/most of our less variables will have names like `@jp-component-css-property` so I would prefer to use a slightly modified naming convention here.

**Comment by ellisonbg** [comment]
This should not close #680 as there are a bunch of other things I need to fix as well...

File: src/basestyle/phosphor.less (hunk 2 above)
Comment (ellisonbg) [inline src/basestyle/phosphor.less]: fixed just now

**Comment by ellisonbg** [comment]
@blink1073 once this has passed Travis again, this should be ready for merge. See above note about less variable names.

**Comment by ellisonbg** [comment]
Merging to keep iterating...

**Pull request merged**